### ISSUE TITLE
Failed to get address for PyRun_SimpleString GetProcAddress: The specified procedure could not be found. [process exited with code 4294967295]

### ISSUE THREAD

**Issue by marios-pz** [pyinstaller/pyinstaller#6332]
Python Ver: 3.10
Pyinstaller and PIP: Up-to-date

I have made a spec file that turns my python project into a single exe with no sub folders, it works great and all
But when I compile the same process over GitHub Actions and then pull my repo to get my executables and open them 
I get this error.
```
Failed to get address for PyRun_SimpleString
GetProcAddress: The specified procedure could not be found.

[process exited with code 4294967295]
```
is it because I use a alpha version of python 3.10 in yaml? 

```
uses: actions/setup-python@v2
        with:
          persist-credentials: false 
          python-version: '3.10.0-alpha.1'
```

**Comment by rokm** [comment]
> is it because I use a alpha version of python 3.10 in yaml?

That does indeed seem to be the case. If I run current `develop` or `v4` branches on CI with python `3.10.0-alpha.1`, it fails with same kind of error ([here](https://github.com/rokm/pyinstaller/actions/runs/1407811766) and [here](https://github.com/rokm/pyinstaller/actions/runs/1407817774)).

It seems that `PyRun_SimpleString` was removed earlier in the python 3.10 development cycle, and re-instated later by [this commit](https://github.com/python/cpython/commit/46bd5ed94cf3d5e03f45eecf9afea1659980c8bf).

So you should be able to avoid the error by simply switching to the released python 3.10.0 (or any pre-release that had that commit merged). 

That said, I'm marking this as a bug as we should probably switch from using the obsolete `PyRun_SimpleString` to `PyRun_SimpleStringFlags`.

**Comment by marios-pz** [comment]
After I did a test repo, I found they released 3.10 behind my back. I am sorry for spreading this miss information , I believe now everything works as it should be. Thanks for the quick support

**Comment by bwoodsend** [comment]
Leave this open. We want to address:
> That said, I'm marking this as a bug as we should probably switch from using the obsolete PyRun_SimpleString to PyRun_SimpleStringFlags.
